crm_compassion/models/weekly_demand.py

```diff
@@ -99,62 +113,108 @@ def _compute_demand_events(self):
             week.number_children_events = allocate
             week.resupply_events = resupply
 
+    @api.model
+    def _default_demand_sub(self):
+        """ Compute average of SUB since one year. """
+        start_date = datetime.today() - timedelta(weeks=STATS_DURATION)
+        sub_sponsored = self.env['recurring.contract'].search_count([
+            ('origin_id.type', '=', 'sub'),
+            ('start_date', '>=', fields.Date.to_string(start_date))
+        ])
+        return float(sub_sponsored) / STATS_DURATION
+
     @api.depends('number_children_website', 'number_children_ambassador',
-                 'number_children_events')
+                 'number_sub_sponsorship', 'number_children_events')
     @api.multi
     def _compute_demand_total(self):
         for week in self:
             week.total_demand = week.number_children_website + \
+                week.number_sub_sponsorship + \
                 week.number_children_ambassador + week.number_children_events
 
     @api.model
     def _default_unsponsored_web(self):
         """ Compute average of unsponsored children on the website since
         one year.
         """
-        start_date = datetime.today() - timedelta(weeks=52)
+        start_date = datetime.today() - timedelta(weeks=STATS_DURATION)
         web_sponsored = self.env['recurring.contract'].search_count([
             ('channel', '=', 'internet'),
             ('start_date', '>=', fields.Date.to_string(start_date)),
-            ('origin_id.type', 'not in', ('partner', 'event', 'sub'))
+            ('origin_id.type', 'not in', ('partner', 'event', 'sub')),
+            ('partner_id.user_ids', '!=', False)
```

Comment:
So I'm sorry but we discussed further about it and this is still wrong. In fact we will count all sponsorships in channel web for the web category. So you can remove the filters of origin and partners.

We will exclude the channel web from the other resupply categories.